### PR TITLE
Provide completion that offers the explicit-interface-type when typing new members.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/18626

This also fixes the linked issue where we weren't offering spell check in explicit-interface-type locations.  That's because spell-check uses 'completion' as the source of information as to what names are valid in what locations.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/Features/CSharp/Portable/Completion/CompletionProviders/CompletionUtilities.cs [hunk 1]
```diff
@@ -80,5 +84,30 @@ public static bool IsStartingNewWord(SourceText text, int characterPosition)
             return CommonCompletionUtilities.IsStartingNewWord(
                 text, characterPosition, IsWordStartCharacter, IsWordCharacter);
         }
+
+        public static (string displayText, string insertionText) GetDisplayAndInsertionText(ISymbol symbol, SyntaxContext context)
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Completion/CompletionProviders/CompletionUtilities.cs]: this is a move.

File: src/Features/CSharp/Portable/Completion/CompletionProviders/CompletionUtilities.cs [hunk 2]
```diff
@@ -80,5 +84,30 @@ public static bool IsStartingNewWord(SourceText text, int characterPosition)
...
+        {
+            var insertionText = GetInsertionText(symbol, context);
+            var displayText = symbol.GetArity() == 0 
+                ? insertionText 
+                : string.Format("{0}<>", insertionText);
+
+            return (displayText, insertionText);
+        }
+
+        public static string GetInsertionText(ISymbol symbol, SyntaxContext context)
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Completion/CompletionProviders/CompletionUtilities.cs]: this is a move.

File: src/Features/Core/Portable/Completion/CommonCompletionProvider.cs [hunk 3]
```diff
@@ -96,5 +97,15 @@ internal virtual bool IsInsertionTrigger(SourceText text, int insertedCharacterP
         {
             return Task.FromResult<TextChange?>(null);
         }
+
+        private static CompletionItemRules s_suggestionItemRules = CompletionItemRules.Create(enterKeyRule: EnterKeyRule.Never);
+
+        protected CompletionItem CreateSuggestionModeItem(string displayText, string description)
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/Completion/CommonCompletionProvider.cs]: these are moves down.

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ExplicitInterfaceTypeCompletionProviderTests.cs [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+            var markup = @"
+using System.Collections;
+
+class C : IList
+{
+    int Foo() => 0;
+    int $$
+
+    [Attr]
+    int Bar();
```
Comment (Pilchie) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ExplicitInterfaceTypeCompletionProviderTests.cs]: Maybe before a type or an accessiblity modifier too?

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ExplicitInterfaceTypeCompletionProviderTests.cs [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+interface I : IList
+{
+    int $$
+}
+";
+
+            await VerifyNoItemsExistAsync(markup);
+        }
+    }
+}
```
Comment (Pilchie) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ExplicitInterfaceTypeCompletionProviderTests.cs]: Verify that it's in suggestion mode?

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ExplicitInterfaceTypeCompletionProviderTests.cs (hunk 4 above)
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ExplicitInterfaceTypeCompletionProviderTests.cs]: sure.

**Pull request merged**